src/data/nat/fib.lean

```diff
@@ -105,17 +105,38 @@ end
 
 /-- See https://proofwiki.org/wiki/Fibonacci_Number_in_terms_of_Smaller_Fibonacci_Numbers -/
 lemma fib_add (m n : ℕ) :
-  fib m * fib n + fib (m + 1) * fib (n + 1) = fib (m + n + 1) :=
+  fib (m + n + 1) = fib m * fib n + fib (m + 1) * fib (n + 1) :=
 begin
   induction n with n ih generalizing m,
   { simp },
   { intros,
     specialize ih (m + 1),
     rw [add_assoc m 1 n, add_comm 1 n] at ih,
-    simp only [fib_add_two, ← ih],
+    simp only [fib_add_two, ih],
     ring, }
 end
 
+lemma fib_two_mul (n : ℕ) : fib (2 * n) = fib n * (2 * fib (n + 1) - fib n) :=
+begin
+  cases n,
+  { simp },
+  { rw [nat.succ_eq_add_one, two_mul, ←add_assoc, fib_add, fib_add_two, two_mul],
+    simp only [← add_assoc, add_tsub_cancel_right],
+    ring, },
+end
+
+lemma fib_two_mul_succ (n : ℕ) : fib (2 * n + 1) = fib (n + 1) ^ 2 + fib n ^ 2 :=
```

Comment:
Should this name be
```suggestion
lemma fib_succ_two_mul (n : ℕ) : fib (2 * n + 1) = fib (n + 1) ^ 2 + fib n ^ 2 :=
```
Is there any precedent one way or another?

Reply:
Mathlib isn't very consistent about this, but I would prefer to write `two_mul_add_one` and reserver `succ` for actual `nat.succ`.

Reply:
Having a pair of `_zero` and `_succ` lemmas is definitely a (nice) pattern across mathlib, but it doesn't apply here. I would therefore use
```suggestion
lemma fib_two_mul_add_one (n : ℕ) : fib (2 * n + 1) = fib (n + 1) ^ 2 + fib n ^ 2 :=
```